site/content/blog/concourse-ytt.md

```diff
@@ -0,0 +1,395 @@
+slug: concourse-ytt-101
+date: 2022-03-08
+author: Neil Hickey
+excerpt: "Tutorial: Concourse to YTT 101"
+tags: ['ytt', 'concourse']
+---
+
+# YTT Tutorial: Converting Concourse pipeline to YTT
+
+## Overview
+
+In today's tutorial I will be giving a quick rundown of the steps I take to convert a simple Concourse yaml file to using the ytt templating framework.
+
+### Pre-requisites
+- ytt (https://carvel.dev/ytt/docs/latest/install/)
+- concourse (https://concourse-ci.org/install.html)
+- docker-compose / docker (optional)
+
+### Getting set up
+
+1. Install a local Concourse server (optional)
+
+```
+wget https://concourse-ci.org/docker-compose.yml && docker-compose up -d
+
+fly --target tutorial login --concourse-url http://localhost:8080 -u test -p test
+```
+
+1. Set up a simple Concourse pipeline
+
+`fly --target tutorial set-pipeline --pipeline testing-pipeline --config pipeline.yml`
+
+<details>
+<summary>Example GoLang Pipeline - pipeline.yml</summary>
+
+```yaml
+---
+resources:
+- name: golang-mock-git
+  type: git
+  icon: github
+  source:
+    uri: https://github.com/golang/mock.git
+
+- name: golang-1.11.x-image
+  type: registry-image
+  icon: docker
+  source:
+    repository: golang
+    tag: 1.11-stretch
+
+- name: golang-1.12.x-image
+  type: registry-image
+  icon: docker
+  source:
+    repository: golang
+    tag: 1.12-stretch
+
+- name: golang-1.13.x-image
+  type: registry-image
+  icon: docker
+  source:
+    repository: golang
+    tag: 1.13-stretch
+
+task-config: &task-config
+  platform: linux
+  inputs:
+    - name: golang-mock-git
+      path: go/src/github.com/golang/mock
+  params:
+    GO111MODULE: "on"
+  run:
+    path: /bin/sh
+    args:
+      - -c
+      - |
+        GOPATH=$PWD/go
+
+        cd go/src/github.com/golang/mock
+
+        go vet ./...
+        go build ./...
+        go install github.com/golang/mock/mockgen
+        GO111MODULE=off go get -u golang.org/x/lint/golint
+        ./ci/check_go_fmt.sh
+        ./ci/check_go_lint.sh
+        ./ci/check_go_generate.sh
+        ./ci/check_go_mod.sh
+        go test -v ./...
+
+jobs:
+- name: golang-1.11
+  public: true
+  plan:
+    - get: golang-mock-git
+      trigger: true
+    - get: golang-1.11.x-image
+      trigger: true
+    - task: run-tests
+      image: golang-1.11.x-image
+      config:
+        << : *task-config
+- name: golang-1.12
+  public: true
+  plan:
+    - get: golang-mock-git
+      trigger: true
+    - get: golang-1.12.x-image
+      trigger: true
+    - task: run-tests
+      image: golang-1.12.x-image
+      config:
+        << : *task-config
+- name: golang-1.13
+  public: true
+  plan:
+    - get: golang-mock-git
+      trigger: true
+    - get: golang-1.13.x-image
+      trigger: true
+    - task: run-tests
+      image: golang-1.13.x-image
+      config:
+        << : *task-config
+```
+</details>
+
+> Taken from https://concourse-ci.org/golang-library-example.html
```

Comment:
What do you think about spending a little time orienting the reader/viewer on what's happening in the pipeline?

I think the investment is worth it given that what makes for good quality extractions is slicing down semantic seams. Emphasizing understanding the meaning of this YAML can help encourage that view.

Reply:
Definitely! Given the length of the post and all the stuff I wanted to fit, this wasn't codified in this PR - but it's definitely valuable to spend that time. Agreed!